cluster-autoscaler/core/scaleup/resource/manager.go

```diff
@@ -180,6 +180,37 @@ func (m *Manager) ApplyLimits(ctx *context.AutoscalingContext, newCount int, res
 	return newCount, nil
 }
 
+// AllocateResources allocates the resources of required number of nodes from resources left and return true if resources are allocated successfully.
+func AllocateResources(left Limits, delta Delta, count int) bool {
```

Comment:
The signature is IMHO a bit misleading - it is not immediately obvious that this method modifies the limits or what is the boolean value representing.

I think something like `func UpdateLimits(left Limits, delta Delta, count int) (Limits, error)` would be better. It also allows to pass the exact error to the call site.